src/mito2/src/flush.rs

```diff
@@ -382,7 +384,31 @@ impl RegionFlushTask {
             timer.stop_and_record(),
         );
 
-        Ok(file_metas)
+        let memtables_to_remove: SmallVec<[_; 2]> = version_data
+            .version
+            .memtables
+            .immutables()
+            .iter()
+            .map(|m| m.id())
+            .collect();
+        let edit = RegionEdit {
+            files_to_add: file_metas,
+            files_to_remove: Vec::new(),
+            compaction_time_window: None,
+            // The last entry has been flushed.
+            flushed_entry_id: Some(version_data.last_entry_id),
+            flushed_sequence: Some(version_data.committed_sequence),
+        };
+        info!("Applying {edit:?} to region {}", self.region_id);
+
+        let action_list = RegionMetaActionList::with_action(RegionMetaAction::Edit(edit.clone()));
+        // We will leak files if the manifest update fails, but we ignore them for simplicity. We can
+        // add a cleanup job to remove them later.
+        self.manifest_ctx
+            .update_manifest(RegionState::Writable, action_list, || {
+                version_control.apply_edit(edit, &memtables_to_remove, self.file_purger.clone());
+            })
+            .await
```

Comment:
Refer to `handle_flush_finished()`
https://github.com/GreptimeTeam/greptimedb/blob/3e9bda326717392945e1b7b4c41864df3ce45517/src/mito2/src/worker/handle_flush.rs#L210-L222